api/src/test/java/org/openmrs/api/impl/ConditionServiceImplTest.java

```diff
@@ -61,13 +61,15 @@ public void saveCondition_shouldSaveNewCondition(){
 		condition.setClinicalStatus(ConditionClinicalStatus.ACTIVE);
 		condition.setUuid(uuid);
 		condition.setPatient(new Patient(patientId));
+		condition.setEncounter(new Encounter(2039));
```

Comment:
Why change an existing test? Is it because this change made it fail?

Reply:
No,  I was just adding this basic case where a condition has an encounter, the `saveCondition(...)` should handle the `encounter` property as expected.

Reply:
When adding new features, changing existing tests creates the impression that the encounter is required and hence existing tests cannot pass without it being added.

Reply:
Sure, I just thought that expanding this existing test is sufficient and appropriate enough for this new property. Wouldn't it be superfluous to have a new test case for this?

Reply:
A new test like conditionShouldSaveWithoutEncounter?

Reply:
I meant it's already covered as is 👍

Reply:
Since you are assigning an encounter to a condition, how does it cover that a condition can save without an encounter?

Reply:
Well, Iet's have a new case were a condition has an encounter and leave the existing test untouched. something like:

```java

public void saveCondition_shouldSaveConditionWithAnEncounter() {
    //...
}

Reply:
But does it make sense to you? 😊

Reply:
Yah it makes sense. 

Thanks!

Reply:
Resolved!

@dkayiwa does the PR look any better? Like mergeable?